feat: add C implementation for `blas/base/ssyr`

aman-095 opened the pull request:
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request proposes to add C implementation for `blas/base/ssyr`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aman-095 committed:
feat: add C implementation for ssyr

aman-095 committed:
chore: resolve lint error

aman-095 committed:
bench: resolve lint errors

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/src/ssyr.c:
```diff
@@ -0,0 +1,80 @@
+		for ( i1 = 0; i1 < N; i1++ ) {
+			if ( X[ ix1 ] != 0.0f ) {
+				tmp = alpha * X[ ix1 ];
+				// oa = (sa1*i1);
```
@aman-095 Should this commented line be here?

aman-095 replied:
No, I forgot to remove it. Done.

aman-095 committed:
chore: remove unnecessary comment

aman-095 committed:
Merge branch 'stdlib-js:develop' into ssyr-c

aman-095 committed:
Merge branch 'stdlib-js:develop' into ssyr-c

aman-095 committed:
refactor: update C implementation for ndarray usage

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
````diff
@@ -183,21 +183,33 @@ console.log( A );
 
 ```c
-TODO
+void c_ssyr( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT N, const float alpha, float *X, const CBLAS_INT strideX, float *A, const CBLAS_INT LDA )
````
```suggestion
void c_ssyr( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT N, const float alpha, const float *X, const CBLAS_INT strideX, float *A, const CBLAS_INT LDA )
```

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
````diff
@@ -183,21 +183,33 @@ console.log( A );
-TODO
+void c_ssyr( const CBLAS_LAYOUT order, const CBLAS_UPLO uplo, const CBLAS_INT N, const float alpha, float *X, const CBLAS_INT strideX, float *A, const CBLAS_INT LDA )
 ```
 
````
@aman-095 Looks like we still need to add the ndarray API docs, correct?

aman-095 replied:
Yes, I will push it up soon.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
```diff
@@ -183,21 +183,33 @@ console.log( A );
+float A[] = { 1.0f, 0.0f, 0.0f, 2.0f, 1.0f, 0.0f, 3.0f, 2.0f, 1.0f };
+float x[] = { 1.0f, 2.0f, 3.0f };
+
+c_ssyr( 102, 121, 3, 1.0f, x, 1, A, 3 );
```
Do not hardcode enum values. Use enum identifiers. The values are subject to change. Enums should be treated as opaque constants.

aman-095 committed:
Merge branch 'stdlib-js:develop' into ssyr-c

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
```diff
@@ -219,7 +231,25 @@ TODO
+int main( void ) {
+    // Create strided arrays:
+    float A[] = { 1.0f, 0.0f, 0.0f, 2.0f, 1.0f, 0.0f, 3.0f, 2.0f, 1.0f };
+    float x[] = { 1.0f, 2.0f, 3.0f };
```
To resolve the CI C lint error, I believe this can be `const float`.

aman-095 committed:
chore: resolve lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aman-095 committed:
docs: update ndarray example in readme

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aman-095 committed:
chore: use enum identifiers

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: resolve lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: add const before variable x

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: add enum identifiers

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: resolve lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: resolve lint error

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aman-095 committed:
chore: add enum identifiers in readme

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
````diff
@@ -183,21 +183,62 @@ console.log( A );
 ```c
-TODO
+#include "stdlib/blas/base/ssyr.h"
+#include "stdlib/blas/base/shared.h"
````
Move this include to the examples, rather than as part of the `ssyr` include code block.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/README.md:
```diff
@@ -219,7 +260,34 @@ TODO
+    float A[] = { 1.0f, 0.0f, 0.0f, 2.0f, 1.0f, 0.0f, 3.0f, 2.0f, 1.0f };
+    const float x[] = { 1.0f, 2.0f, 3.0f };
+
+    // Specify the number of elements:
```
```suggestion
    // Specify the number of elements along each dimension of `A`:
```

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,190 @@
+* Runs a benchmark.
+*
+* @param iterations   number of iterations
+* @param len          array length
+* @return elapsed time in seconds
```
```suggestion
* @param len          array length
* @return                elapsed time in seconds
```

This has been updated throughout the code base to be aligned now. Applies below, as well.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,190 @@
+*
+* @param iterations   number of iterations
+* @param len          array length
+* @return elapsed time in seconds
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
Would you mind renaming to just `benchmark.c`? Length is not an appropriate descriptor. Size is more appropriate, as we are dealing with matrices, not vectors. In this case, we don't need to differentiate the file name, so fine to just name `benchmark.c`.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,190 @@
+* Runs a benchmark.
+*
+* @param iterations   number of iterations
+* @param len          array length
```
```suggestion
* @param N            number of elements along each dimension
```

Applies below, as well. `len` is more appropriate for vectors, not matrices.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,190 @@
+	print_version();
+	count = 0;
+	for ( i = MIN; i <= MAX; i++ ) {
+		len = stdlib_base_floorf( pow( pow( 10, i ), 1.0/2.0 ) );
```
```suggestion
		N = stdlib_base_floorf( pow( pow( 10, i ), 1.0/2.0 ) );
```

Affects here, below, and above.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.length.c:
```diff
@@ -0,0 +1,190 @@
+		iter = ITERATIONS / pow( 10, i-1 );
+		for ( j = 0; j < REPEATS; j++ ) {
+			count += 1;
+			printf( "# c::%s:len=%d\n", NAME, len );
```
```suggestion
			printf( "# c::%s:size=%d\n", NAME, N*N );
```

Applies below, as well.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/examples/c/example.c:
```diff
@@ -0,0 +1,46 @@
+	float A[] = { 1.0f, 0.0f, 0.0f, 2.0f, 1.0f, 0.0f, 3.0f, 2.0f, 1.0f };
+	const float x[] = { 1.0f, 2.0f, 3.0f };
+
+	// Specify the number of elements:
```
Same comments as README.

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/manifest.json:
```diff
@@ -0,0 +1,319 @@
+        "@stdlib/blas/base/shared"
+      ]
+    },
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/src/ssyr.c:
```diff
@@ -0,0 +1,131 @@
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
+* @param N        number of elements along each dimension of `A`
+* @param alpha    scalar
+* @param x        input vector
+* @param strideX  `x` stride length
```
```suggestion
* @param X        input vector
* @param strideX  `X` stride length
```

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/src/ssyr.c:
```diff
@@ -0,0 +1,131 @@
+* @param N         number of elements along each dimension of `A`
+* @param alpha     scalar
+* @param x         input vector
+* @param strideX   `x` stride length
+* @param offsetX   starting index of `x`
```
```suggestion
* @param strideX   `X` stride length
* @param offsetX   starting index of `X`
```

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/src/ssyr_cblas.c:
```diff
@@ -0,0 +1,37 @@
+* @param N        number of elements along each dimension of `A`
+* @param alpha    scalar
+* @param x        input vector
+* @param strideX  `x` stride length
```
```suggestion
* @param strideX  `X` stride length
```

kgryte commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into ssyr-c

aman-095 committed:
Merge branch 'stdlib-js:develop' into ssyr-c

aman-095 commented:
/stdlib update-copyright-years

aman-095 committed:
chore: apply review changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aman-095 committed:
bench: remove unutilized function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,180 @@
+#include "stdlib/blas/base/ssyr.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/blas/ext/base/sfill.h"
+#include "stdlib/math/base/special/floorf.h"
```
```suggestion
```

kgryte committed:
bench: remove include

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/ssyr/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,179 @@
+	print_version();
+	count = 0;
+	for ( i = MIN; i <= MAX; i++ ) {
+		N = stdlib_base_floorf( pow( pow( 10, i ), 1.0/2.0 ) );
```
```suggestion
		N = floor( pow( pow( 10, i ), 1.0/2.0 ) );
```

kgryte committed:
bench: use built-in

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
